src/main/groovy/com/palantir/gradle/gitversion/JGitDescribe.groovy

```diff
@@ -0,0 +1,101 @@
+package com.palantir.gradle.gitversion
+
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.Constants
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.PersonIdent
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevTag
+import org.eclipse.jgit.revwalk.RevWalk
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class JGitDescribe implements GitDescribe {
+    private static final Logger log = LoggerFactory.getLogger(JGitDescribe.class)
+
+    private File directory
+
+    JGitDescribe(File directory) {
+        this.directory = directory
+    }
+
+    @Override
+    String describe(String prefix) {
+        Git git = Git.wrap(new FileRepository(GitCli.getRootGitDir(directory)))
+        if (!GitUtils.isBackCompatible(git)) {
```

Comment:
Not a huge fan of the method name here... why not just call this `isRepoEmpty` ?

Reply:
Makes sense. Let's not log anything